lib/common.sh

```diff
@@ -789,6 +789,45 @@ common::print_n_char () {
 }
 
 ###############################################################################
+# Generate a (github) markdown TOC between BEGIN/END tags
+# @param file The file to update in place
+#
+common::mdtoc () {
+  local file=$1
+  local indent
+  local anchor
+  local heading
+  local begin_block="<!-- BEGIN MUNGE: GENERATED_TOC -->"
+  local end_block="<!-- END MUNGE: GENERATED_TOC -->"
+  local tmpfile=/tmp/$PROG-cm.$$
+
+  declare -A count
+
+  while read level heading; do
+    indent="$(echo $level |sed -e "s,^#,,g" -e 's,#,  ,g')"
+    # make a valid anchor
+    anchor=${heading,,}
+    anchor=${anchor// /-}
+    anchor=${anchor//[\.\?\*\,\/()]/}
+    # Keep track of dups and identify
+    ((count[$anchor]++)) ||true
+    ((${count[$anchor]}>1)) && anchor+="-${count[$anchor]}"
```

Comment:
this line is wrong.

Looking at https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG.md#client-binaries, the Client Binaries anchor is https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG.md#client-binaries, followed by https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG.md#client-binaries-1, etc.

per the logic on https://github.com/jch/html-pipeline/blob/master/lib/html/pipeline/toc_filter.rb#L43-L44, probably need to increment count after we use it.